### PR TITLE
MC-Packet Expansion for improved streaming support

### PR DESCRIPTION
**Pull request by Salanto**

### CODE REVIEW

File: src/aomusicplayer.cpp [hunk 1]
```diff
@@ -13,12 +13,12 @@ AOMusicPlayer::~AOMusicPlayer()
...
 }
 
-void AOMusicPlayer::play(QString p_song, int channel, bool loop,
+int AOMusicPlayer::play(QString p_song, int channel, bool loop,
                          int effect_flags)
 {
   channel = channel % m_channelmax;
   if (channel < 0) // wtf?
-    return;
+    return 18;
```
Comment (oldmud0) [inline src/aomusicplayer.cpp:21]: what is 18

**Review by oldmud0 — COMMENTED** [review]
I don't understand what the improvements are. You want to fall back to a content URL when the asset isn't found...?

**Comment by Salanto** [comment]
> I don't understand what the improvements are. You want to fall back to a content URL when the asset isn't found...?

If a user does not have the content in their base folder, it would try to fall back to a content URL link before showing the file as missing. 
As most major servers already have webao repositories, it would be a small thing to set up serverside.  Akashi, for example, has a PR just waiting to be merged to implement it.

The implementation is not bulletproof, I admit that, but it's using the currently implemented code as best it can without complicating the AO2 Client further.

It also shows, unlike the current client, when a remote file is missing, instead of using the content URL to determine it is a stream.

The PR is mostly aimed to address the massive amount of contents some server supply to their userbase, in excess of 10GB sometimes, which has a significant amount of music files that are not being played and end up being wasted disk space. By offering a streaming solution that uses the music list the user and server owner have an easier option to trade file size with bandwidth usage. It's invisible to those who have the files and can be hidden on the theme for users who prefer to stream music.

**Comment by oldmud0** [comment]
Add a packet that suggests a remote content base. We can use it in webAO as well.

**Comment by Salanto** [comment]
Closed. Different implementation based on #548 will follow.

**Pull request closed without merging**